src/System.Management.Automation/engine/remoting/commands/NewPSSessionConfigurationFile.cs

```diff
@@ -15,6 +15,7 @@
 
 namespace Microsoft.PowerShell.Commands
 {
+#if !UNIX
     /// <summary>
     /// New-PSSessionConfigurationFile command implementation
```

Comment:
Maybe split the file by classes and exclude a new file from compile on Unix?

If we look CodeFactor issues it recommends to use one-class-per-file - if we move code in the PR maybe follow this rule?

Reply:
Once-class-per-file is ridiculous. We have modern editors with all kinds of features for managing large files. I'm much happier navigating within a few files than having to deal with hundreds of tabs. I vote we suppress this rule.

Reply:
Agree with @BrucePay in the general case of having multiple classes per file. Sometimes it's overkill to make a class nested, but it makes a lot more sense contextually to have it in the same file as a larger class. Enums are a good motivating example I find. Also, it's nice to feel like I've moved on from this historical holdover of Java and MATLAB...

Reply:
My question was only about the file. The rule wasalready disabled in PR #6930.

Reply:
@iSazonov The reason I was using `#if !UNIX` in this file is because the class `SessionConfigurationUtils` in this file is heavily used and only used in those two cmdlets (`New-PSSessionConfigurationFile` and `New-PSRoleCapabilityFile`), so it makes sense to keep them close to each other in the same file.